linker/jvm/src/test/scala/org/scalajs/linker/RunTest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker
+
+import scala.concurrent._
+
+import org.junit.{Rule, Test}
+import org.junit.Assert._
+import org.junit.rules.TemporaryFolder
+
+import org.scalajs.ir.Trees._
+
+import org.scalajs.junit.async._
+
+import org.scalajs.jsenv._
+import org.scalajs.jsenv.nodejs.NodeJSEnv
+
+import org.scalajs.linker.interface._
+import org.scalajs.linker.testutils._
+import org.scalajs.linker.testutils.LinkingUtils._
+import org.scalajs.linker.testutils.TestIRBuilder._
+
+class RunTest {
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  @(Rule @scala.annotation.meta.getter)
+  val tempFolder = new TemporaryFolder()
+
+  @Test
+  def jlClassIsCore_Issue4616(): AsyncResult = await {
+    // Check that if jl.Class is instantiated, it must be depended upon by jl.Object.
+
+    val classDefs = Seq(
+      mainTestClassDef({
+        consoleLog(ClassOf(T))
+      })
+    )
+
+    val linkerConfig = StandardConfig()
+      .withModuleKind(ModuleKind.ESModule)
+      .withModuleSplitStyle(ModuleSplitStyle.SmallestModules)
+      .withSourceMap(false)
+      .withOutputPatterns(OutputPatterns.fromJSFile("%s.mjs"))
+
+    val output = tempFolder.newFolder().toPath
+
+    val env = new NodeJSEnv
+
+    testLink(classDefs, MainTestModuleInitializers,
+        linkerConfig, PathOutputDirectory(output)).flatMap { _ =>
+      val run = env.start(Seq(Input.ESModule(output.resolve("main.mjs"))), RunConfig())
+
+      run.future.finallyWith(Future(run.close()))
+    }
```

Comment:
Looks like this section is a good candidate to be extracted in a method `testLinkdAndRun` or something like that, doesn't it?

Reply:
Indeed. I'll probably use the `TestKit` from `JSEnv` so that it can take care of all the I/O handling and what not (otherwise, we'll spam the test stdout).